src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/convert_bf16_fp32_fp16.cpp

```diff
@@ -0,0 +1,146 @@
+    std::cout << "entering " << model.get_runtime_model()->get_ops().size() << std::endl;
+    for ( const auto& n : model.get_runtime_model()->get_ordered_ops() ) {
+        std::cout << "znaleziono " << n->get_friendly_name() << std::endl;
+        if ( n->get_friendly_name() == "conversion_bf16_to_f32" ) {
+            ASSERT_EQ(n->get_input_element_type(0),  ov::element::bf16);
+            ASSERT_EQ(n->get_output_element_type(0),  ov::element::f32);
+            convert_found = true;
+        }
+    }
+    ASSERT_TRUE(convert_found);
+}
+*/
+TEST_F(BF16WeightsDecompression, Inference_input_3) {
+    std::shared_ptr<ov::Model> function = init_subgraph();
+    ov::CompiledModel cM = core->compile_model(function, targetDevice, {ov::hint::inference_precision(ov::element::f32)});
+    auto request = cM.create_infer_request();
+    request.infer();
+    auto out_tensor = request.get_output_tensor();
+    for ( size_t i = 0 ; i < out_tensor.get_size(); i++ ) {
+        ASSERT_FLOAT_EQ(reinterpret_cast<float*>(out_tensor.data())[i], 12);
+    }
+}
+
+TEST_F(BF16WeightsDecompression, Inference_input_3_f16) {
+    std::shared_ptr<ov::Model> function = init_subgraph();
+    ov::CompiledModel cM = core->compile_model(function, targetDevice, {ov::hint::inference_precision(ov::element::f16)});
+    auto request = cM.create_infer_request();
+    request.infer();
+    auto out_tensor = request.get_output_tensor();
+    for ( size_t i = 0 ; i < out_tensor.get_size(); i++ ) {
+        ASSERT_FLOAT_EQ(reinterpret_cast<float*>(out_tensor.data())[i], 12);
+    }
+}
+
+TEST_P(BF16WeightsDecompression, Inference_without_convert) {
+    float given;
+    float expected;
+    std::tie(given, expected) = GetParam();
+    std::shared_ptr<ov::Model> function_without_convert = init_subgraph_without_convert(given);
+    ov::CompiledModel model_without_convert = core->compile_model(function_without_convert, targetDevice, {ov::hint::inference_precision(ov::element::f32)});
+    auto request = model_without_convert.create_infer_request();
+    request.infer();
+    auto out_tensor = request.get_output_tensor();
+    for ( size_t i = 0 ; i < out_tensor.get_size(); i++ ) {
+        ASSERT_NEAR(reinterpret_cast<unsigned short*>(out_tensor.data())[i], expected, 0.2f); //3072
+    }
+}
+
+INSTANTIATE_TEST_SUITE_P(Inference_without_convert, BF16WeightsDecompression,
+                         testing::Values(BF16WeightsDecompressionParams(768, 17728.0f),
+                            BF16WeightsDecompressionParams(3.0f, 16704.0f),
+                            BF16WeightsDecompressionParams(2.0f, 16640.0f),
+                            BF16WeightsDecompressionParams(49512.0f, 18497.0f), //768x64
+                            BF16WeightsDecompressionParams(98304.0f, 18624.0f), //768x128
+                            BF16WeightsDecompressionParams(55340232221128654848.0f, 24896.0f),
+                            BF16WeightsDecompressionParams(-55340232221128654848.0f, 57664.0f),
```

Comment:
are these numbers generated by reinterpreting bf16 values? Then what about having an explicit function to convert/reinterpreting function?

Reply:
I don't think It is good idea to check function call by another function call

Reply:
but I found that way to check in framework, what should I do?

Reply:
the problem with current code is that it is difficult to understand the meaning of these numbers. Then what about adding more comments about these numbers?

Reply:
I think that I will add one function, then no long description needed